packages/core/core-flows/src/order/workflows/order-edit/update-order-edit-item-quantity.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  OrderChangeActionDTO,
+  OrderChangeDTO,
+  OrderDTO,
+  OrderPreviewDTO,
+  OrderWorkflow,
+} from "@medusajs/types"
+import { ChangeActionType, OrderChangeStatus } from "@medusajs/utils"
+import {
+  WorkflowData,
+  WorkflowResponse,
+  createStep,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { useRemoteQueryStep } from "../../../common"
+import {
+  previewOrderChangeStep,
+  updateOrderChangeActionsStep,
+} from "../../steps"
+import {
+  throwIfIsCancelled,
+  throwIfOrderChangeIsNotActive,
+} from "../../utils/order-validation"
+
+/**
+ * This step validates that an item can be updated from an order edit.
+ */
+export const updateOrderEditItemQuantityValidationStep = createStep(
+  "update-order-edit-update-quantity-validation",
+  async function (
+    {
+      order,
+      orderChange,
+      input,
+    }: {
+      order: OrderDTO
+      orderChange: OrderChangeDTO
+      input: OrderWorkflow.UpdateOrderEditItemQuantityWorkflowInput
+    },
+    context
+  ) {
+    throwIfIsCancelled(order, "Order")
+    throwIfOrderChangeIsNotActive({ orderChange })
+
+    const associatedAction = (orderChange.actions ?? []).find(
+      (a) => a.id === input.action_id
+    ) as OrderChangeActionDTO
+
+    if (!associatedAction) {
+      throw new Error(
+        `No request to update item quantity for order ${input.order_id} in order change ${orderChange.id}`
+      )
+    } else if (associatedAction.action !== ChangeActionType.ITEM_UPDATE) {
+      throw new Error(`Action ${associatedAction.id} is not updating an item`)
+    }
+  }
+)
+
+export const updateOrderEditItemQuantityWorkflowId =
+  "update-order-edit-update-quantity"
+/**
+ * This workflow updates a new item in the order edit.
+ */
+export const updateOrderEditItemQuantityWorkflow = createWorkflow(
```

Comment:
q: isn't this workflow just edit item and not scoped to qty?

Reply:
The update item quantity is a new action recently added where we can set any arbitrary number to it.
We had remove_item and add_item before, but they behave differently.

Reply:
I meant, just calling this `updateOrderEditItemWorkflow` as we're updating both note and quantity.

Reply:
I see. That makes sense, I will rename it.